microsoft365/errors.go

```diff
@@ -32,7 +33,11 @@ func getErrorObject(err error) *RequestError {
 		}
 	}
 
-	return nil
+	// If the type of error is other than ODataError
+	// return the exact error
+	return &RequestError{
+		Message: err.Error(),
```

Comment:
Are we able to return other properties from the error, like its code, or do we only have the message consistently?

Reply:
@cbruno10 Generally, the errors are of type `*odataerrors.ODataError` or `*azidentity.AuthenticationFailedError`.
https://github.com/microsoftgraph/msgraph-sdk-go#31-get-the-users-drive
But if the error type is different (not sure what are the other common error types), we need to cast the error to get the proper error code.
For now, we are returning the entire API error as the message.